firefox-ios/Client/Frontend/Settings/HomepageSettings/HomePageSettingViewController.swift

```diff
@@ -128,23 +124,25 @@ class HomePageSettingViewController: SettingsTableViewController, FeatureFlaggab
             format: .Settings.Homepage.CustomizeFirefoxHome.ThoughtProvokingStoriesSubtitle,
             PocketAppName.shortName.rawValue)
 
-        let jumpBackInSetting = BoolSetting(
-            with: .jumpBackIn,
-            titleText: NSAttributedString(string: .Settings.Homepage.CustomizeFirefoxHome.JumpBackIn)
-        ) { value in
-            store.dispatch(
-                JumpBackInAction(
-                    isEnabled: value,
-                    windowUUID: self.windowUUID,
-                    actionType: JumpBackInActionType.toggleShowSectionSetting
-                )
-            )
-        }
-
         // Section ordering
         sectionItems.append(TopSitesSettings(settings: self))
 
-        if isJumpBackInSectionEnabled {
+        if let profile {
+            let jumpBackInSetting = BoolSetting(
+                prefs: profile.prefs,
+                theme: themeManager.getCurrentTheme(for: windowUUID),
+                prefKey: PrefsKeys.UserFeatureFlagPrefs.JumpBackInSection,
+                defaultValue: true,
+                titleText: .Settings.Homepage.CustomizeFirefoxHome.JumpBackIn
+            ) { value in
+                store.dispatch(
+                    JumpBackInAction(
+                        isEnabled: value,
+                        windowUUID: self.windowUUID,
+                        actionType: JumpBackInActionType.toggleShowSectionSetting
+                    )
+                )
+            }
             sectionItems.append(jumpBackInSetting)
         }
 
```

Comment:
We have another `if let profile {` on the line below, we could combine both `jumpBackInSetting` and `bookmarksSetting` in the same `if` so we only unwrap the profile once?